### ISSUE TITLE
Allow null on swagge rspec when allow_nil is true

### ISSUE THREAD

**Issue by nullnull** [Apipie/apipie-rails#650]
`SwaggerGenerator#swagger_atomic_param` method gets `allow_nulls` argument and manage types of the parameter, but I think we could not pass `allow_nulls = true` to this method.

How can I pass `allow_nulls = true` to the method?
And I think It'd be better to use `allow_nil` parameter to manage null types (simple example is here https://github.com/nullnull/apipie-rails/pull/1/commits/421b27d542f1001ba12135cf812f748d5f5844be), What do you think?

**Comment by iNecas** [comment]
I agree we should use `allow_nil` at all places we currently hardcode the `allow_null` value. it should always come from `param_desc`

**Comment by mathieujobin** [comment]
anyone else encountered an issue with apipie when upgrading to rails 5 related to nil becoming empty strings? especially during tests?
